### DISCUSSION TITLE
How to synchronize changes applied to the Magento2 code from Windows11?

### DISCUSSION THREAD

**Discussion by devenairevo** [markshust/docker-magento#984] [Q&A]
Hey guys,

how to synchronize changes applied to the Magento2 code from Windows11/WSL2/Docker?

Magento2 is running fine on Docker, and Windows11 with WSL2, but after the changes are applied in the code Magento2 does not update the changes, seems that the Docker containers are not aware that the changes to the local project were applied.

Maybe someone had a similar issue?

Thanks.

**Comment by kennylawrence** [comment]
Are you using PhpStorm or VSCode? As far as I know, these IDE's if configured properly are what perform the sync between the WSL filesystem and the Docker containers. Generally, in the IDE you have to:

* create a php cli interpreter of type "Docker Compose" (this is one of the types you can select) that points to the YML config files in your WSL host - it will use this interpreter to issue commands inside the phpfpm container (I think)
* specify the phpfpm service of the Docker multi-container (it gets this info from the YML files, so you just select phpfpm)
* specify /usr/local/bin/php as the PHP binary inside the phpfpm container
* make a path mapping between the src folder on the wsl host (ie, \\wsl.localhost\Ubuntu\home\kenny\Sites\magento\src ) and the one in the phpfpm container at /var/www/html
* and finally select the new php interpreter for use in your project; all is done at File->Settings->PHP in PhpStorm

If you haven't done it already, the above is covered in this part of the Docker setup course: https://courses.m.academy/courses/487758/lectures/9748834

I'm using PhpStorm, so not sure about setting up VSCode. But you know it is working properly if the project panel shows one directory tree for your WSL host's copy of magento and another below that inside of External Libraries/PHP/src -- and, if you create a file or folder in one of the trees, it will also show up in its corresponding location in the other one.

The latest version of phpstorm is a little bit different than that used in the video, but you can figure it out. It took me a few tries to push through when things didn't look right.. just keep exploring when you get stuck. I've heard there is a new version of the course coming out before too long that will use the latest phpstorm version as a reference.

**Comment by RowBowl** [comment]
To cover the vscode side of things, please see: https://code.visualstudio.com/docs/remote/wsl-tutorial and check to see if your vscode is connected to wsl (bottom left as seen in the screenshot) and remotely connect if not. If all the windows related configurations are set up, it should work smoothly, i've had no issues seeing my changes applied through wsl instantly (unless the change requires cache clean)
